mlir/test/lib/Dialect/Test/TestTypes.cpp

```diff
@@ -505,6 +505,10 @@ Type TestRecursiveAliasType::parse(AsmParser &parser) {
     return rec;
   }
 
+  // Allow incomplete definitions that can be completed later.
+  if (succeeded(parser.parseGreater()))
```

Comment:
```suggestion
  if (succeeded(parser.parseOptionalGreater()))
```
Surprised this doesn't cause any issues I must admit.